PROTOCOL.md

```diff
@@ -253,6 +253,47 @@ as of now. The add and remove file actions are stored as their individual column
 
 These files reside in the `_delta_log/_sidecars` directory.
 
+### Log Compaction Files
```

Comment:
Missing ToC entry:
![image](https://github.com/delta-io/delta/assets/79601771/f444bb90-4b10-420a-a28b-18c780797668)

Reply:
We should probably expand [Metadata cleanup](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#metadata-cleanup) section to cover log compaction as well?